commit 919bad841d338f5efd4fbca119439c1308b9bd83
Author: bertin <macportsuser@macports.org>
Date:   Fri Oct 10 19:16:20 2025 +0200

    /opt/local/linux-ports/audio/audacity/files/patch-enable-compact-menu.diff

diff --git a/src/menus/FileMenus.cpp b/src/menus/FileMenus.cpp
index 2161d37fd..cd19335b7 100644
--- a/src/menus/FileMenus.cpp
+++ b/src/menus/FileMenus.cpp
@@ -247,6 +247,7 @@ void OnClose(const CommandContext &context )
 void OnCompact(const CommandContext &context)
 {
    ProjectFileManager::Get(context.project).Compact();
+   UndoManager::Get(context.project).ClearStates();
 }
 
 void OnSave(const CommandContext &context )
@@ -495,15 +496,15 @@ auto FileMenu()
                AudioIONotBusyFlag() ),
             Command( wxT("SaveCopy"), XXO("&Backup Project..."), OnSaveCopy,
                AudioIONotBusyFlag() )
-         )//,
+         ),
 
          // Bug 2600: Compact has interactions with undo/history that are bound
          // to confuse some users. We don't see a way to recover useful amounts
          // of space and not confuse users using undo.
          // As additional space used by aup3 is 50% or so, perfectly valid
          // approach to this P1 bug is to not provide the 'Compact' menu item.
-         //Command( wxT("Compact"), XXO("Co&mpact Project"), OnCompact,
-         //   AudioIONotBusyFlag(), wxT("Shift+A") )
+         Command( wxT("Compact"), XXO("Co&mpact Project (invalidates Undo/History commands!)"), OnCompact,
+           AudioIONotBusyFlag() )
       ),
 
       Section( "Close", Command( wxT("Close"), XXO("&Close"), OnClose,
